doc/articles/contributing/guidelines/code-style.md

```diff
@@ -0,0 +1,87 @@
+# Guidelines for Code style
+
+There are a mixture of patterns within the codebase of Uno and in the interest to ensure the correct patterns proliferate this page serves as the projects north star. When you are resolving a bug or feature please refactor the surrounding code thus [leave the code cleaner than you found it](https://www.matheus.ro/2017/12/11/clean-code-boy-scout-rule/) whilst keeping the changes minimal:
+
+> Some developers, when faced with fixing, or adding a feature to an open source project are under the mistaken impression that the first step before any fixing takes place, or before adding a new feature takes place is to make the code "easier for them" to work on.
+>
+> "Easier for them" usually is a combination of renaming methods, fields, properties, locals; Refactoring of methods, classes; Gratuitous split of code in different files, or merging of code into a single file; Reorganization by alphabetical order, or functional order, or grouping functions closer to each other, or having helper methods first, or helper methods last. Changing indentation, aligning variables, or parameters or dozen other smaller changes.
+>
+> This is *not how you contribute to an open source project*.
+>
+> When you contribute fixes or new features to an open source project you should use the existing coding style, the existing coding patterns and stick by the active maintainer's choice for his code organization.
+>
+> The maintainer is in for the long-haul, and has been working on this code for longer than you have. Chances are, he will keep doing this even after you have long moved into your next project.
+>
+> Sending a maintainer a patch, or a pull request that consists of your "fix" mixed with a dozen renames, refactoring changes, variable renames, method renames, file splitting, layout changing code is not really a contribution, it is home work.
+>
+> The maintainer now has to look at your mess of a patch and extract the actual improvement, wasting precious time that could have gone to something else. This sometimes negates the effort of your "contribution".
+>
+> *Open Source Contribution Etiquette by Miguel de Icaza - https://tirania.org/blog/archive/2010/Dec-31.html*
+
+
+Uno uses EditorConfig ([here's our configuration](https://github.com/unoplatform/uno/blob/master/.editorconfig)) to maintain consistent coding styles and settings in our codebase, such as indent style, tab width, end of line characters, encoding, and more. We typically observe the [Microsoft C# Coding Conventions](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/inside-a-program/coding-conventions) with one notable exception - Uno uses Tabs. If you install this [Visual Studio plugin](https://marketplace.visualstudio.com/items?itemName=mynkow.FormatdocumentonSave) it will automatically format your contributions upon file save.
```

Comment:
> We typically observe the Microsoft C# Coding Conventions.

Requires fact checking. Is there already an existing document within nventive that we can open-source and incorporate into this document?

Reply:
There isn't one that would be fit for this, but what you've described fits well.